pkg/server/asset/s3.go

```diff
@@ -136,17 +137,18 @@ func (s *s3Store) PutFileReader(
 // GeneratePostFileRequest return a PostFileRequest for uploading asset
 func (s *s3Store) GeneratePostFileRequest(name string, contentType string, length int64) (*PostFileRequest, error) {
 	return &PostFileRequest{
-		Action: "/files/" + name,
+		Action: "/files/" + url.PathEscape(name),
```

Comment:
I am not sure whether we should pass the path-escaped key when we interact with the Go SDK of AWS. The SDK may expect unescaped key.

Line 140 and line 149 definitely need manual escape because they are not interacting with the AWS SDK.

Reply:
Second thought. If line 140 used to have unescaped URL, file with space still can be uploaded successfully. Maybe we need only to change line 149.

Reply:
Why not merge into `v1-branch`?

Reply:
Seems like `v1-branch` is identical to `v1.7.3`. Sorry, forgot what I just said.

Reply:
I don't think the Go AWS SDK handles key escaping, the whole repo does not use `url.PathEscape` once.

Reply:
> Seems like v1-branch is identical to v1.7.3. Sorry, forgot what I just said.

I was thinking we should release a `v1.7.4`.

Reply:
> the whole repo does not use url.PathEscape once.

We did use `url.PathEscape` in `pkg/server/asset/cloud.go` but we did not use Go AWS SDK there so no conclusion can be made. 

> I don't think the Go AWS SDK handles key escaping

Then I wonder why we can upload file with name having spaces. The observed problem now is incorrectly generated URL, which should be related to line 149.

Sorry I am a bit conservative because we do not have tests to verify our changes :(

Reply:
> We did use url.PathEscape in pkg/server/asset/cloud.go but we did not use Go AWS SDK there so no conclusion can be made.

I meant the AWS Go SDK repo, not ours.

Reply:
Not sure if it is related. https://github.com/aws/aws-sdk-go/issues/1789